Xamarin.PropertyEditing.Mac/Controls/CombinablePropertyEditor.cs

```diff
@@ -0,0 +1,144 @@
+using System;
+using System.Collections;
+using System.Diagnostics;
+using System.ComponentModel;
+
+using Foundation;
+using AppKit;
+using CoreGraphics;
+
+using Xamarin.PropertyEditing.ViewModels;
+using Xamarin.PropertyEditing.Mac.Resources;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Xamarin.PropertyEditing.Mac
+{
+	internal class CombinablePropertyEditor<T>
+		: PropertyEditorControl
+	{
+		public CombinablePropertyEditor ()
+		{
+			base.TranslatesAutoresizingMaskIntoConstraints = false;
+
+			UpdateTheme ();
+		}
+
+		public override NSView FirstKeyView => firstKeyView;
+		public override NSView LastKeyView => lastKeyView;
+
+		internal new CombinablePropertyViewModel<T> ViewModel
+		{
+			get { return (CombinablePropertyViewModel<T>)base.ViewModel; }
+			set { base.ViewModel = value; }
+		}
+
+		List<NSButton> combinableList = new List<NSButton> ();
+		bool dataPopulated;
+		NSView firstKeyView;
+		NSView lastKeyView;
+
+		protected override void HandleErrorsChanged (object sender, DataErrorsChangedEventArgs e)
+		{
+			UpdateErrorsDisplayed (ViewModel.GetErrors (e.PropertyName));
+		}
+
+		protected override void SetEnabled ()
+		{
+			foreach (var item in combinableList) {
+				item.Enabled = ViewModel.Property.CanWrite;
+			}
+		}
+
+		protected override void UpdateErrorsDisplayed (IEnumerable errors)
+		{
+			if (ViewModel.HasErrors) {
+				SetErrors (errors);
+			} else {
+				SetErrors (null);
+				SetEnabled ();
+			}
+		}
+
+		protected override void OnViewModelChanged (PropertyViewModel oldModel)
+		{
+			if (!dataPopulated) {
+
+				combinableList.Clear ();
+
+				const float controlHeight = 22;
+
+
+				// Set our new RowHeight
+				RowHeight =  (ViewModel.Choices.Count () * controlHeight);
+
+				float top = controlHeight;
+				foreach (var item in ViewModel.Choices) {
+					var BooleanEditor = new NSButton (new CGRect (4, RowHeight - top, Frame.Width - 4, controlHeight)) {
+						AllowsMixedState = true,
+						ControlSize = NSControlSize.Small,
+						Font = NSFont.FromFontName (DefaultFontName, DefaultFontSize),
+						Title = item.Name,
+						TranslatesAutoresizingMaskIntoConstraints = false,
+					};
+					BooleanEditor.SetButtonType (NSButtonType.Switch);
+					BooleanEditor.Activated += SelectionChanged;
+
+					AddSubview (BooleanEditor);
+					combinableList.Add (BooleanEditor);
+					top += controlHeight;
+				}
+
+				// Set our tabable order
+				firstKeyView = combinableList.First ();
+				lastKeyView = combinableList.Last ();
+
+				dataPopulated = true;
+			}
+
+			base.OnViewModelChanged (oldModel);
+		}
+
+		protected override void UpdateValue ()
+		{
+			foreach (var item in combinableList) {
+				if (ViewModel.Choices.Count () > 0) {
```

Comment:
`Any()` is faster than `Count() > 0`. Beyond that, `Choices` is an `IReadOnlyList`, we don't need to go to LINQ to see if it's empty.

Reply:
I hadn't see the Count property on IReadOnlyList. Much nicer.